File: 2022-2023/sourcecode/lab/lab3/WfiAppJava/app/src/main/java/pl/edu/uwr/pum/wfiappjava/InstituteAdapter.java
package pl.edu.uwr.pum.wfiappjava;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;

import java.util.ArrayList;

public class InstituteAdapter extends RecyclerView.Adapter<InstituteAdapter.InstituteViewHolder> {

    public static final String INSTITUTE_EXTRA = "pl.edu.uwr.pum.wfiappjava.institute";

    private final ArrayList<Institute> institutes;
    private final Context context;

    public InstituteAdapter(Context context, ArrayList<Institute> instituteData){
        this.institutes = instituteData;
        this.context = context;
    }

    public class InstituteViewHolder extends RecyclerView.ViewHolder implements View
            .OnClickListener{

        private final TextView titleTextView;
        private final TextView infoTextView;
        private final ImageView instituteImage;

        public InstituteViewHolder(@NonNull View itemView) {
            super(itemView);

            titleTextView = itemView.findViewById(R.id.title);
            infoTextView = itemView.findViewById(R.id.subTitle);
            instituteImage = itemView.findViewById(R.id.instituteImage);

            itemView.setOnClickListener(this);
        }

        public void bind(Institute currentInstitute){
            titleTextView.setText(currentInstitute.getTitle());
            infoTextView.setText(currentInstitute.getInfo());

            Glide.with(context).load(currentInstitute.getImageResource())
                    .into(instituteImage);
        }

        @Override
        public void onClick(View v) {
            Institute currentInstitute = institutes.get(getAdapterPosition());
            Intent intent = new Intent(context, DetailActivity.class);
            intent.putExtra(INSTITUTE_EXTRA, currentInstitute);
            context.startActivity(intent);
        }
    }

    @NonNull
    @Override
    public InstituteAdapter.InstituteViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new InstituteViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.rv_item,
                parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull InstituteAdapter.InstituteViewHolder holder, int position) {
        Institute currentInstitute = institutes.get(position);
        holder.bind(currentInstitute);
    }

    @Override
    public int getItemCount() {
        return institutes.size();
    }
}

File: 2022-2023/sourcecode/lab/lab3/WFiAppKotlin/app/src/main/java/pl/edu/uwr/pum/wfiappkotlin/InstituteAdapter.kt
package pl.edu.uwr.pum.wfiappkotlin
import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide

const val INSTITUTE_EXTRA = "pl.edu.uwr.pum.wfiappkotlin.institute"

class InstituteAdapter(
    private val context: Context,
    private val institutes: ArrayList<Institute>) :
    RecyclerView.Adapter<InstituteAdapter.InstituteViewHolder>() {

    inner class InstituteViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        private val titleTextView: TextView = itemView.findViewById(R.id.title)
        private val infoTextView: TextView = itemView.findViewById(R.id.subTitle)
        private val instituteImage: ImageView = itemView.findViewById(R.id.instituteImage)

        init {
            itemView.setOnClickListener(this)
        }

        fun bind(currentInstitute: Institute) {
            titleTextView.text = currentInstitute.title
            infoTextView.text = currentInstitute.info

            Glide.with(context).load(currentInstitute.imageResource)
                .into(instituteImage)
        }

        override fun onClick(p0: View?) {
            val institute = institutes[adapterPosition]

            val intent = Intent(context, DetailActivity::class.java).apply {
                putExtra(INSTITUTE_EXTRA, institute)
            }

            context.startActivity(intent)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): InstituteViewHolder =  InstituteViewHolder(
            LayoutInflater.from(parent.context).inflate(
                R.layout.rv_item,
                parent, false
            )
        )

    override fun onBindViewHolder(holder: InstituteViewHolder, position: Int) {
        val currentInstitute = institutes[position]
        holder.bind(currentInstitute)
    }

    override fun getItemCount() = institutes.size
}